src/esm.ts

```diff
@@ -28,7 +29,12 @@ export function registerAndCreateEsmHooks(opts?: RegisterOptions) {
     preferTsExts: tsNodeInstance.options.preferTsExts,
   });
 
-  return { resolve, getFormat, transformSource };
+  // The hooks API changed in node version X so we need to check for backwards compatibility
+  // TODO: When the new API is released, change to the correct node version here
+  const newHooksAPI = parseInt(process.versions.node.split('.')[0], 10) >= 17;
+  return newHooksAPI
+    ? { resolve, load }
+    : { resolve, getFormat, transformSource };
```

Comment:
Somewhere in this file we should add a couple comments explaining our approach, since it might throw off a new reader.  We are implementing the *new* hooks API on top of our implementation of the *old* hooks API, which relies on copy-pasted code from the *old* hooks implementation in node.

Reply:
Added some comments in `load` but perhaps more are needed?

Reply:
Thanks, I made some nitpicky changes, mostly I added a big block comment to the top of the file explaining today's implementation and how we will likely change it in the future.  I think (I hope?) this should address any confusion.